### PR TITLE
 commit 07.02.2019

### PR DESCRIPTION
**Pull request by MateAcademy**
my homework 07.02.2019

### CODE REVIEW

File: src/main/resources/Untitled2.sql [hunk 1]
```diff
@@ -0,0 +1,150 @@
+--Homework 07.02.2019:
+SELECT DISTINCT COMPANY FROM ORDERS o1 INNER JOIN PRODUCTS p1 ON o1.PRODUCT = p1.product_id INNER JOIN CUSTOMERS c1 ON o1.cust = c1.cust_num
```
Comment (u2107680007911543774e7r) [inline src/main/resources/Untitled2.sql]: how would you select if you need all the information and not just a name of a company?

File: src/main/resources/Untitled2.sql [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+    (
+        SELECT
+            *
+        FROM
+            salesreps
+            INNER JOIN offices ON salesreps.quota > offices.target
+    )
+WHERE
+    quota > target;
+
```
Comment (u2107680007911543774e7r) [inline src/main/resources/Untitled2.sql]: copy-past? :)
you have different style...

File: src/main/resources/Untitled2.sql [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+SELECT AVG(100 * (SALES/QUOTA)) FROM SALESREPS;
+
+SELECT AVG(100 * (SALES/QUOTA)) PROCENT FROM SALESREPS;
+
+SELECT SUM(QUOTA), SUM(SALES) FROM SALESREPS;
+
+SELECT SUM(AMOUNT) FROM ORDERS t1
+INNER JOIN SALESREPS t2  ON t1.REP =t2.EMPL_NUM
+WHERE NAME = '���� �������';
```
Comment (u2107680007911543774e7r) [inline src/main/resources/Untitled2.sql]: ? : )

File: src/main/resources/Untitled2.sql [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+            salesreps
+            INNER JOIN (
+                SELECT
+                    city,
+                    office
+                FROM
+                    offices
+            ) ON salesreps.rep_office = office
+    );
+--
```
Comment (u2107680007911543774e7r) [inline src/main/resources/Untitled2.sql:272]: same copy-past... but it's ok if you can understand how it works

**Review by u2107680007911543774e7r — APPROVED** [review]
Acceptable...
You need to format all the code, not just the parts that you copied : )
On you machine it's Ctrl+A, Ctrl-f7 I guess.

File: src/main/resources/Untitled2.sql (hunk 1 above)
Comment (SerhiiPasko) [reply src/main/resources/Untitled2.sql]: It is ok. But if you use alias for table names use it for columns too. c1.COMPANY in your case

File: src/main/resources/Untitled2.sql [hunk 5]
```diff
@@ -0,0 +1,150 @@
+--Homework 07.02.2019:
+SELECT DISTINCT COMPANY FROM ORDERS o1 INNER JOIN PRODUCTS p1 ON o1.PRODUCT = p1.product_id INNER JOIN CUSTOMERS c1 ON o1.cust = c1.cust_num
+WHERE p1.DESCRIPTION LIKE '%a%' OR p1.DESCRIPTION LIKE '%A%' OR p1.description LIKE '%�%'  OR p1.description LIKE '%�%';
+--
+SELECT ORDER_NUM, AMOUNT, COMPANY, CREDIT_LIMIT  FROM ORDERS, CUSTOMERS  WHERE CUST = CUST_NUM;
```
Comment (SerhiiPasko) [inline src/main/resources/Untitled2.sql]: As clean code write join, where from new line

File: src/main/resources/Untitled2.sql [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+--    
+SELECT NAME, SALESREPS.SALES, CITY	FROM SALESREPS, OFFICES	WHERE REP_OFFICE = OFFICE;
+SELECT NAME, t1.sales,  CITY	FROM OFFICES t2  INNER JOIN  SALESREPS t1 ON t1.rep_office=t2.office;
+SELECT
+    name,
+    sales,
+    city
+FROM
+    (
+        SELECT
```
Comment (SerhiiPasko) [inline src/main/resources/Untitled2.sql]: Why you use internal select in this case? This is don't need

**Review by SerhiiPasko — CHANGES_REQUESTED** [review]
Read comments. Critical for me - internal selects

File: src/main/resources/Untitled2.sql (hunk 2 above)
Comment (MateAcademy) [reply src/main/resources/Untitled2.sql]: Yes, this option is different, interesting for me

File: src/main/resources/Untitled2.sql (hunk 3 above)
Comment (MateAcademy) [reply src/main/resources/Untitled2.sql]: I saw it, when I poured it onto the githab, it happened, but in the task it was written: WHERE NAME = 'Дима Маликов';

File: src/main/resources/Untitled2.sql (hunk 4 above)
Comment (MateAcademy) [reply src/main/resources/Untitled2.sql:272]: Yes, I figured out this example, there is a personal decision from above

File: src/main/resources/Untitled2.sql (hunk 6 above)
Comment (MateAcademy) [reply src/main/resources/Untitled2.sql]: This is a very interesting solution, only because

**Review by SerhiiPasko — APPROVED** [review]
Good

**Pull request merged**